src/toxcore/friend_connection/packet/share_relays.rs

```diff
@@ -0,0 +1,82 @@
+/*! ShareRalays struct
+*/
+
+use toxcore::binary_io::*;
+use toxcore::state_format::old::*;
+
+/** ShareRelays is a struct that holds 3 TCP relays as a PackedNode format.
+
+This packet is used to share relay info. between two friends.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x11`
+variable  | 3 PackedNodes
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ShareRelays {
+    relays: Vec<TcpUdpPackedNode>,
```

Comment:
Is it possible that this packet contains UDP addresses? If not, then can we create an alternative PackedNode but for TCP addresses only?

Reply:
Or maybe just add different methods for serialize/deserialize PacketNode with TCP and with UDP address?

Reply:
There are two options.
PackedNode of UDP, TCP, TCP_UDP, 3 separated struct is one option.
One struct which has different methods for TCP and UDP and TcpUdp.
One struct way seems better, but this case, other part of source code should be changed.

Reply:
But is seems you need only TCP addresses here, right? Then it seems to me that it's better to reuse PackedNode and change/add parsing/serializing methods.

Reply:
Ok.
I will try to do that you said.

Reply:
I looked through our state format implementation - it seems we don't need `TcpUdpPackedNode` there as well - we know at compile time what type should be used (no need to fix it now if anything).